src/js/modules/users/reducer.js

```diff
@@ -32,9 +32,30 @@ const reducer = (state = { ...initialState }, action) => {
         }, {}),
       };
     }
-    case FETCH_USER_REJECTED: {
+    case FETCH_USERS_REJECTED: {
       return { ...state, isFetching: false, error: action.payload };
     }
+    case FETCH_USER_ROLES_FULFILLED: {
+      return { ...state, userRoles: action.payload };
+    }
+    case FETCH_ROLES_FULFILLED: {
+      return {
+        ...state,
+        roles: action.payload.reduce((acc, role) => {
+          acc[role.id] = role;
+          return acc;
+        }, {}),
+      };
+    }
+    case CREATE_USER_FULFILLED: {
+      return {
+        ...state,
+        users: {
+          ...state.users,
+          [action.payload.data.id]: action.payload.data,

```

Comment:
You were in the middle of adding this when I took your laptop. Make sure it's right? I think it is but idk

Reply:
Yup. It actually broke when I first got it, but I added `.data` and now it works.